analyzers/tests/SonarAnalyzer.UnitTest/TestCases/VariableShadowsField.CSharp12.cs

```diff
@@ -0,0 +1,14 @@
+﻿class MyClass(int first, int second, int third)
+{
+    private int first; // Compliant
+
+    void MyMethod()
+    {
+        int second = 1; // Compliant
+
+        if (first is object third) // Compliant
+        {
+            return;
+        }
```

Comment:
The body of this `if` doesn't seem relevant for the test, and the non-compliance assertion can be made a one-liner:
```cs
_ = first is object third; // Compliant
```